src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1318,61 +1318,94 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
             VisitCondition(node.Condition);
             var consequenceState = this.StateWhenTrue;
             var alternativeState = this.StateWhenFalse;
+
+            var consequence = VisitConditionalOperand(consequenceState, node.Consequence, isByRef);
+            Unsplit();
+            consequenceState = this.State;
+            var consequenceResult = _result;
+
+            var alternative = VisitConditionalOperand(alternativeState, node.Alternative, isByRef);
+            Unsplit();
+            var alternativeResult = _result;
+
+            bool? getIsNullable(Result result) => result.Type is null ? true : result.Type.IsNullable;
+            bool? resultIsNullable;
             if (IsConstantTrue(node.Condition))
             {
-                VisitConditionalOperand(alternativeState, node.Alternative, isByRef);
-                VisitConditionalOperand(consequenceState, node.Consequence, isByRef);
-                // it may be a boolean state at this point.
+                SetState(consequenceState);
+                resultIsNullable = getIsNullable(consequenceResult);
             }
             else if (IsConstantFalse(node.Condition))
             {
-                VisitConditionalOperand(consequenceState, node.Consequence, isByRef);
-                VisitConditionalOperand(alternativeState, node.Alternative, isByRef);
-                // it may be a boolean state at this point.
+                resultIsNullable = getIsNullable(alternativeResult);
             }
             else
             {
-                VisitConditionalOperand(consequenceState, node.Consequence, isByRef);
-                Unsplit();
-                var consequenceType = _result.Type;
-                consequenceState = this.State;
-                VisitConditionalOperand(alternativeState, node.Alternative, isByRef);
-                Unsplit();
-                var alternativeType = _result.Type;
                 IntersectWith(ref this.State, ref consequenceState);
-                if ((object)consequenceType == null || (object)alternativeType == null || !consequenceType.Equals(alternativeType, TypeCompareKind.ConsiderEverything))
+                resultIsNullable = (getIsNullable(consequenceResult) | getIsNullable(alternativeResult));
+            }
+
+            TypeSymbolWithAnnotations resultType;
+            if (consequence.Kind == BoundKind.ThrowExpression)
```

Comment:
>consequence.Kind == BoundKind.ThrowExpression [](start = 16, length = 45)

It is not clear why throw expression needs special handling. It doesn't look like binding handles it specially.